src/main/kotlin/com/manimdsl/shapes/Shapes.kt

```diff
@@ -1,15 +1,108 @@
 package com.manimdsl.shapes
 
-interface Shape {
+import com.manimdsl.linearrepresentation.Alignment
 
-    val text: String
-    val classPath: String
-    val className: String
-    val pythonVariablePrefix: String
+sealed class Shape {
+    abstract val ident: String
+    abstract val text: String
+    abstract val classPath: String
+    abstract val className: String
+    abstract val pythonVariablePrefix: String
+    val style = Style()
+
+    open fun getConstructor(): String {
+        return "$ident = ${className}(\"${text}\"$style)"
+    }
+
+    override fun toString(): String {
+        return "$ident.all"
+    }
 }
 
-data class Rectangle(override val text: String) : Shape {
+sealed class ShapeWithText : Shape() {
+    // Python shape classes which has shape and text members to manipulate color
+
+    fun getTextMObject(): String {
+        return "$ident.text"
+    }
+
+    fun getShapeMObject(): String {
+        return "$ident.shape"
+    }
+}
+
+class Rectangle(
+    override val ident: String,
+    override val text: String,
+    private val color: String? = null,
+    private val textColor: String? = null,
+    private val textWeight: String? = null,
+    private val font: String? = null,
+) : ShapeWithText() {
     override val classPath: String = "python/rectangle.py"
     override val className: String = "Rectangle_block"
     override val pythonVariablePrefix: String = "rectangle"
+
+    init {
+        color?.let { style.addStyleAttribute(Color(color)) }
+        textColor?.let { style.addStyleAttribute(TextColor(textColor)) }
+        textWeight?.let { style.addStyleAttribute(TextWeight(textWeight)) }
+        font?.let { style.addStyleAttribute(Font(font)) }
```

Comment:
Are all these let's necessary? You don't seem to use "it" inside the lambda?

Reply:
Oh I can replace it with it :/

Reply:
No but if it's working without using it can you not just call `style.addStyleAttribute(Color(color)) ` without the let?

Reply:
@arj119

Reply:
Oh no as if it is null it should not add anything

Reply:
The defaults are in the python side not Kotlin. Otherwise the constructors generated in the main construct function will be messy

Reply:
Ah okay cool looks good then